IBM/carbon-addons-iot-react
# 1262 table cells will overflow under some conditions

Closes #1262

**Summary**
The renderDataFunction is not included in the CellRenderer which means that apps that use renderDataFunction will miss out on CSS settings causing cell text wrapping and truncation to fail where it could have worked.

**Note**
Note sure if this should be considered breaking functionality since cells that are using renderDataFunction now could be wrapped with these classes  (when the wrapText prop is used) which might affect that cell rendering.

```
.#{$iot-prefix}--table__cell-text--truncate {
  @include text-overflow();
  display: block;
}

.#{$iot-prefix}--table__cell-text--no-wrap {
  white-space: nowrap;
}

```

**Change List (commits, features, bugs, etc)**
- Moved renderDataFunction into TableCellRenderer so that the cell from a renderDataFunction will be wrapped by the same CSS as a normal cell
- Added a unit test to catch the original issue

**Acceptance Test (how to verify the PR)**
- Check that the unit test looks correct and that it is passing

There is no story that shows this particular bug, but you can check the following:

1. Go to https://deploy-preview-1286--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-table--stateful-example-with-expansion-maxpages-and-column-resize
2. select 'auto' in the knob `wrapCellText``
3. the status column uses renderDataFunction so you can check by inspecting the DOM that the SVG is wrapped in a span with class `iot--table__cell-text--truncate`
4. Also check that this story still looks okay https://deploy-preview-1286--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-table--with-custom-cell-renderer
